lib/onelogin/ruby-saml/idp_metadata_parser.rb

```diff
@@ -66,33 +79,65 @@ def get_idp_metadata(url, validate_cert)
         meta_text
       end
 
+      # @return [String|nil] IdP Entity ID value if exists
+      #
       def idp_entity_id
-        node = REXML::XPath.first(document, "/md:EntityDescriptor/@entityID", { "md" => METADATA })
+        node = REXML::XPath.first(
```

Comment:
Do all of these value accessors guarantee that the value was fetched from a part of the SAML response that was covered by the XMLDSIG signature? There have been successful attacks on SAML implementations by sending assertions with additional fields that were not covered by the signature, but were trusted because the parts of the assertion that were signed were valid.